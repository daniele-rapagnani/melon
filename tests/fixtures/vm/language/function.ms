
func test => {
    io.print("Void function");
}

func test2 |x| => {
    io.print("First arg: " .. string.toString(x));
}

func test3 |x, y| => {
    io.print("First arg: " .. string.toString(x));
    io.print("Second arg: " .. string.toString(y));
    return x .. y;
}

function.call(test, []);
function.call(test2, ["test"]);
function.call(test2, []);
function.call(test2, ["test", "test2"]);
io.print(function.call(test3, ["test", "test2"]));

io.print(function.getName(test3));
io.print(function.getName(test));
io.print(function.getName(func asd => {}));
io.print(function.getArgsCount(test));
io.print(function.getArgsCount(test2));
io.print(function.getArgsCount(test3));

io.print(function.getFile(test3));
io.print(function.getFile(test3) == function.getFile(test));