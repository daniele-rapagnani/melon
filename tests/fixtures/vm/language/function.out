Void function
First arg: test
First arg: null
First arg: test
First arg: test
Second arg: test2
testtest2
test3
test
asd
0
1
2
null
true
